src/middlewared/middlewared/plugins/vm/supervisor/supervisor_base.py

```diff
@@ -274,14 +274,51 @@ def get_features_xml(self):
             features.append(
                 create_element('kvm', attribute_dict={'children': [create_element('hidden', state='on')]})
             )
+
+        if self.vm_data['hyperv_enlightenments']: 
+            features.append(self.get_hyperv_xml())
+        
         return create_element(
             'features', attribute_dict={
                 'children': [
                     create_element('acpi'),
+                    create_element('apic'),
```

Comment:
@sonicaj what do you think, can adding this break anything for the existing users?

Reply:
I'm by no means expert on this topic but how I implemented this feature is opt-in so by default it doesn't have any effect on existing users. Also it allows users to try if their virtualization stack support these features by simply checking the hyper-v checkbox in UI. If their machine deosn't support the features then the UI will prompt error message like "Unable start VM foobar, CPU doesn't support featuate XYZ", or something similar (I verfied this on my SCALE instance by enabling some new flags not included in this review). After that user can simply uncheck the hyper-v checkbox and machine will start like it used to.

I'm open to adjusting the default set on enlightenments but this seemed like a reasonable default set which I have experimented with. Also it's important to remember that this doesn't take away anything from existing user, only enables better Windows support for those who are able to use it.

Reply:
`<apic/>` will be added to `<features>` for all VMs, does not matter if enlightenments are enabled or not, that's why I am asking.

Reply:
Oh sorry my bad. Based on how the libvirt's documentation is worded I would say it shouldn't:

> APIC allows the use of programmable IRQ management.

So it should only enable things, not force.
https://libvirt.org/formatdomain.html#hypervisor-features

Reply:
We should be good i believe